src/core/meshcop/commissioner.cpp

```diff
@@ -389,7 +391,10 @@ otError Commissioner::RemoveJoiner(const Mac::ExtAddress *aEui64, uint32_t aDela
 
             joiner->mValid = false;
             UpdateJoinerExpirationTimer();
-            SendCommissionerSet();
+            if (aNotifyLeader)
+            {
+                SendCommissionerSet();
```

Comment:
Long term, I think it is more appropriate to pull this out of `RemoveJoiner()`. Doing so would allow for cases where multiple joiners are removed at the same time. In that case, only one message needs to be sent.

Reply:
I was thinking about using a new private method like `RemoveJoinerImpl` for only modifying the `mJoiners`. But later I was convinced by myself to only add a bool argument.

Reply:
Agree it would be better to aggregate `MGMT_COMMISSIONER_SET.req` for the addition/removal of multiple joiners into one.

Some options in my mind are
1) split the `MGMT_COMMISSIONER_SET.req` from Add/Remove Joiner operation, specifically
    -  no MGMT_COMMISSIONER_SET.req when adding/removing Joiner;
    -  introduce one API like `otCommissionerDatasetRegister()` which will call `Commissioner::SendCommissionerSet(void)` to construct one MGMT_COMMISSIONER_SET.req

     The side effect is that we should add note for the user that this extra API should be explicitly called to let the addition/removal of joiner to take effect.

2)  introduce some delay for `MGMT_COMMISSIONER_SET.req` per addition/removal operation, similar as existing aggregation solution for Child Update Request.

Thoughts?

Reply:
I think splitting the functionalities of one `commissioner joiner add` command into multiple commands make things unnecessarily more complex for users. 
I think 2) is a good idea, which we can enhance accordingly in another separate PR.